internal/engine/compiler/impl_amd64.go

```diff
@@ -274,19 +274,23 @@ func (c *amd64Compiler) compileGlobalGet(o *wazeroir.OperationGlobalGet) error {
 	// When an integer, reuse the pointer register for the value. Otherwise, allocate a float register for it.
 	valueReg := intReg
 	var vt runtimeValueType
-	var inst = amd64.MOVQ
+	var inst asm.Instruction
 	switch c.ir.Globals[o.Index].ValType {
 	case wasm.ValueTypeI32:
+		inst = amd64.MOVL
 		vt = runtimeValueTypeI32
 	case wasm.ValueTypeI64, wasm.ValueTypeExternref, wasm.ValueTypeFuncref:
+		inst = amd64.MOVQ
 		vt = runtimeValueTypeI64
 	case wasm.ValueTypeF32:
+		inst = amd64.MOVL
 		vt = runtimeValueTypeF32
 		valueReg, err = c.allocateRegister(registerTypeVector)
 		if err != nil {
 			return err
 		}
 	case wasm.ValueTypeF64:
+		inst = amd64.MOVQ
```

Comment:
maybe add default panic BUG because this requires a close look in rare case a new value type